src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -74,319 +75,297 @@ const defaultProps = {
     preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
 };
 
-class ReportActionItemMessageEdit extends React.Component {
-    constructor(props) {
-        super(props);
-        this.updateDraft = this.updateDraft.bind(this);
-        this.deleteDraft = this.deleteDraft.bind(this);
-        this.debouncedSaveDraft = _.debounce(this.debouncedSaveDraft.bind(this), 1000);
-        this.publishDraft = this.publishDraft.bind(this);
-        this.triggerSaveOrCancel = this.triggerSaveOrCancel.bind(this);
-        this.onSelectionChange = this.onSelectionChange.bind(this);
-        this.addEmojiToTextBox = this.addEmojiToTextBox.bind(this);
-        this.setExceededMaxCommentLength = this.setExceededMaxCommentLength.bind(this);
-        this.saveButtonID = 'saveButton';
-        this.cancelButtonID = 'cancelButton';
-        this.emojiButtonID = 'emojiButton';
-        this.messageEditInput = 'messageEditInput';
-
-        let draftMessage;
-        if (this.props.draftMessage === this.props.action.message[0].html) {
+// native ids
+const saveButtonID = 'saveButton';
+const cancelButtonID = 'cancelButton';
+const emojiButtonID = 'emojiButton';
+const messageEditInput = 'messageEditInput';
+
+function ReportActionItemMessageEdit(props) {
+    const [draft, setDraft] = useState(() => {
+        if (props.draftMessage === props.action.message[0].html) {
             // We only convert the report action message to markdown if the draft message is unchanged.
             const parser = new ExpensiMark();
-            draftMessage = parser.htmlToMarkdown(this.props.draftMessage).trim();
-        } else {
-            // We need to decode saved draft message because it's escaped before saving.
-            draftMessage = Str.htmlDecode(this.props.draftMessage);
+            return parser.htmlToMarkdown(props.draftMessage).trim();
         }
-
-        this.state = {
-            draft: draftMessage,
-            selection: {
-                start: 0,
-                end: 0,
-            },
-            isFocused: false,
-            hasExceededMaxCommentLength: false,
-        };
-    }
-
-    componentDidMount() {
+        // We need to decode saved draft message because it's escaped before saving.
+        return Str.htmlDecode(props.draftMessage);
+    });
+    const [selection, setSelection] = useState({start: 0, end: 0});
+    const [isFocused, setIsFocused] = useState(false);
+    const [hasExceededMaxCommentLength, setHasExceededMaxCommentLength] = useState(false);
+
+    const textInputRef = useRef(null);
+    const isFocusedRef = useRef(false);
```

Comment:
Instead of doing this:

```js
const [isFocused, setIsFocused] = useState(false);
// Use this ref to keep track of the previous value of `isFocused`
const isFocusedRef = useRef(false);
useEffect(() => {
    isFocusedRef.current = isFocused;
}, [isFocused])
```

Let's instead use the custom `usePrevious` hook in `src/hooks`. It's a pretty common pattern and does this same thing, but is a bit clearer in it's intention imo:

```js
const [isFocused, setIsFocused] = useState(false);
const wasFocused = usePrevious(isFocused);
```